overlord/configstate/configcore/homedirs.go

```diff
@@ -0,0 +1,195 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nomanagers
+// +build !nomanagers
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sandbox/apparmor"
+)
+
+// For mocking in tests
+var (
+	osOpenFile = os.OpenFile
+	osStat     = os.Stat
+
+	apparmorUpdateHomedirsTunable        = apparmor.UpdateHomedirsTunable
+	apparmorLoadProfiles                 = apparmor.LoadProfiles
+	apparmorSnapConfineDistroProfilePath = apparmor.SnapConfineDistroProfilePath
+)
+
+var (
+	// No path located under any of these top-level directories can be used.
+	// This is not a security measure (the configuration can only be changed by
+	// the system administrator anyways) but rather a check around unintended
+	// mistakes.
+	invalidPrefixes = []string{
+		"/bin/",
+		"/boot/",
+		"/dev/",
+		"/etc/",
+		"/lib/",
+		"/proc/",
+		"/root/",
+		"/sbin/",
+		"/sys/",
+		"/tmp/",
+	}
+
+	// TODO: remove this once we mount the root FS of a snap as tmpfs and
+	// become able to create any mount points. But for the time being, let's
+	// allow only those locations that can be supported without creating mimics
+	validPrefixes = []string{
+		"/home/",
+	}
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.homedirs"] = true
+}
+
+func updateHomedirsConfig(config string) error {
+	configPath := filepath.Join(dirs.SnapdStateDir(dirs.GlobalRootDir), "system.info")
```

Comment:
I think system-params (also matching the style of system-key) might be a better name for this